components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java

```diff
@@ -325,32 +326,33 @@ private String processLogoutRequest(HttpServletRequest request, HttpServletRespo
         String redirectURL = null;
         Cookie opBrowserStateCookie = OIDCSessionManagementUtil.getOPBrowserStateCookie(request);
         String idTokenHint = request.getParameter(OIDCSessionConstants.OIDC_ID_TOKEN_HINT_PARAM);
+        String clientId = request.getParameter(OIDCSessionConstants.OIDC_CLIENT_ID_PARAM);
         String postLogoutRedirectUri = request
                 .getParameter(OIDCSessionConstants.OIDC_POST_LOGOUT_REDIRECT_URI_PARAM);
         String state = request
                 .getParameter(OIDCSessionConstants.OIDC_STATE_PARAM);
 
-        String clientId;
-        String appTenantDomain = null;
+        String appTenantDomain;
         try {
-            if (OIDCSessionManagementUtil.isIDTokenEncrypted(idTokenHint)) {
-                appTenantDomain = request.getParameter(OIDCSessionConstants.OIDC_TENANT_DOMAIN_PARAM);
-                JWT decryptedIDToken = OIDCSessionManagementUtil.decryptWithRSA(appTenantDomain, idTokenHint);
-                clientId = OIDCSessionManagementUtil.extractClientIDFromDecryptedIDToken(decryptedIDToken);
-            } else {
-                if (!validateIdToken(idTokenHint)) {
-                    String msg = "ID token signature validation failed.";
-                    if (log.isDebugEnabled()) {
-                        log.debug(msg);
+            if (StringUtils.isEmpty(clientId)) {
```

Comment:
```suggestion
            if (StringUtils.isBlank(clientId)) {
```

Reply:
Check other relevant places and switch to isBlank as well

Reply:
addressed with [054f072](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2169/commits/054f07256bc9967336c3e5c7d9b3315ab85184b2)